include/irr/asset/IAssetLoader.h

```diff
@@ -7,38 +7,104 @@
 namespace irr { namespace asset
 {
 
+//! A class automating process of loading Assets from resources, eg. files
+/**
+	Every Asset must be loaded by a particular class derived from IAssetLoader.
+	These classes must be registered with IAssetManager::addAssetLoader() which will 
+	add it to the list of loaders (grab return 0-based index) or just not register 
+	the loader upon failure (don�t grab and return 0xdeadbeefu).
+
+	The loading is impacted by caching and resource duplication flags, defined as IAssetLoader::E_CACHING_FLAGS.
+
+	There are defined rules of loading process, that can be overwritten, but basically
+	a mesh can reference a submesh, a submesh can reference a material, a material 
+	can reference a texture, etc. You can look at it as recursive function, the LEVEL (\bhierarchyLevel\b)
+	is actually how deep the stack you are. The flag is a bitfield with 2 bits per level,
```

Comment:
This needs specifying a bit @AnastaZIuk, finally the level is defined as number of -> chains between root (the one that shall be returned by `IAssetManager::getAsset` to the user) and the currently loaded asset.

Stuff that comes together from a single resource (file) like a Mesh->MeshBuffer->GraphicsPipeline->Shaders will only be cached at the highest level local to the file (the root relative to file) so in this case the `Mesh`.

Reply:
do you want add something to these docs because I'm not sure?

Reply:
nope @AnastaZluk only.

Reply:
@AnastaZIuk would you be so kind and update the docs?

Reply:
@AnastaZIuk ?

Reply:
I would for sure, I'm sorry I didn't see it then

Reply:
@devshgraphicsprogramming done